web/app/controllers/application_controller.rb

```diff
@@ -2,4 +2,9 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  rescue_from CanCan::AccessDenied do |exception|
+    flash[:error] = exception.message
```

Comment:
I don't know enough about Rails to know what this `flash[:error]` does. pls splain? This like a modal thing?